okhttp/third_party/okhttp/java/io/grpc/okhttp/internal/Platform.java

```diff
@@ -60,6 +62,16 @@
 public class Platform {
   public static final Logger logger = Logger.getLogger(Platform.class.getName());
 
+  /**
+   * List of security providers to use in order of preference.
+   */
+  private static final String[] ANDROID_SECURITY_PROVIDERS = new String[]{
+      // See https://developer.android.com/training/articles/security-gms-provider.html
+      "com.google.android.gms.org.conscrypt.OpenSSLProvider",
+      "com.android.org.conscrypt.OpenSSLProvider",
+      "org.conscrypt.OpenSSLProvider",
```

Comment:
If these are in preference order, I would imagine this would be first in the list, since it is being provided explicitly by the application. But really, I'm not seeing too much reason to avoid the preference order returned from Security.getProviders(): "The order of the providers in the array is their preference order." Yes, we know that some of these are better than others, but if an application has elected to put something lower in the list, then that was their choice and they likely have their reasons for doing so.